var/spack/repos/builtin/packages/valgrind/package.py

```diff
@@ -69,11 +73,11 @@ def configure_args(self):
         if not (spec.satisfies('%clang') and sys.platform == 'darwin'):
             # Otherwise with (Apple's) clang there is a linker error:
             # clang: error: unknown argument: '-static-libubsan'
-            options.append('--enable-ubsan')
+            if spec.satisfies('+ubsan'):
```

Comment:
Can you mention the cases where this has any effect? I think it would be cleaner to put a `conflicts()` rather than silently ignore variants the user demanded

Reply:
On macOS High Sierra with Clang 6.0, the ubsan library needed to be disabled for valgrind to build.

Reply:
I am unfamiliar with conflicts().